room/room.go

```diff
@@ -127,6 +129,19 @@ func (t *TalkRoom) ReceiveMessages(ctx context.Context) (chan ocs.TalkRoomMessag
 			if err != nil {
 				continue
 			}
+
+			// If it seems that we no longer have access to the chat, stop the goroutine and set error in the next return.
+			if res.StatusCode == 404 {
+				_ = res.Body.Close()
+				c <- ocs.TalkRoomMessageData{Error: ErrRoomNotFound}
+				return
+			}
+			if res.StatusCode == 401 {
+				_ = res.Body.Close()
+				c <- ocs.TalkRoomMessageData{Error: ErrUnauthorized}
+				return
+			}
```

Comment:
```suggestion
			}
			if res.StatusCode == 429 {
				_ = res.Body.Close()
				c <- ocs.TalkRoomMessageData{Error: ErrTooManyRequests}
				return
			}
```

Reply:
Let's go with this for now and open an issue for maybe having a config for wait on 429?

Reply:
Well 429 happens when you try to brute force login credentials or your token expired and you therefor use invalid credentials

Reply:
My bad. In that case, let's just quit.